chatrooms.lua

```diff
@@ -94,13 +145,39 @@ function ToArrival( tUser, sMessage )
 		return true
 	else
 		local sChat = sMessage:match "%b$$(.*)|"
-		SendToRoom( tUser, sTo, sChat )
+		SaveToFile( sChat, sTo )
+		local sCmd, sData = sMessage:match "%b$$%b<>%s+[-+*/?!#](%w+)%s?(.*)|"
+		if not sCmd then
+			SendToRoom( tUser, sTo, sChat )
+		elseif sCmd:lower() == "history" then
+			local sReply, iLimit = "Past %d messages: \n\n\t%s\n\n", tonumber( sData )
+			if (not iLimit) or iLimit > 35 or iLimit < 0 then iLimit = 15 end
+			sReply = sReply:format( iLimit, History(iLimit, sTo) )
+			Core.SendPmToUser( tUser, sTo, sReply )
+		elseif sCmd:lower() == "l" or sCmd:lower == "list" then
+			local sTemplate = ("There are %d current users allowed here:\n\n\t"):format( #(tChatRooms[sTo].tUsers) )
```

Comment:
The reason behind using a template for reply was so that program will not allocate memory to it in every execution. Define it somewhere in larger scope.